				} catch (Exception e) {
					SubTaskFailureException stfe = newSubTaskFailureException(task, e);
					throw stfe;
				}

				return result;
			}

		});
		taskQueue.put(ft);
